packages/forms/backend/sirius-components-forms-graphql/src/main/java/org/eclipse/sirius/components/forms/graphql/datafetchers/form/TreeNodeEndIconURLDataFetcher.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.forms.graphql.datafetchers.form;
+
+import java.util.List;
+import graphql.schema.DataFetchingEnvironment;
+
+import org.eclipse.sirius.components.annotations.spring.graphql.QueryDataFetcher;
+import org.eclipse.sirius.components.forms.TreeNode;
+import org.eclipse.sirius.components.graphql.api.IDataFetcherWithFieldCoordinates;
+import org.eclipse.sirius.components.graphql.api.URLConstants;
+
+/**
+ * Data fetcher for TreeNode.iconEndURL, to rewrite the relative path of the image into an absolute path on the server.
+ * <p>
+ * If the <code>TreeNode.iconEndURL</code> is of the form <code>path/to/image.svg</code>, the rewritten value which will
+ * be seen by the frontend will be <code>/api/images/path/to/image.svg</code>.
+ *
+ * @author mcharfadi
+ */
+@QueryDataFetcher(type = "TreeNode", field = "iconEndURL")
```

Comment:
It seems you've forgotten to also update the name of the field here to `endIconsURL`.